modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractAdaCodegen.java

```diff
@@ -426,10 +426,8 @@ public CodegenOperation fromOperation(String path, String httpMethod, Operation
         final Map<String, List<String>> scopes = new HashMap<>();
         for (SecurityRequirement requirement : securities) {
             for (String key : requirement.keySet()) {
-                SecurityScheme securityScheme = securitySchemes.get(key);
-                if (securityScheme != null) {
-                    scopes.put(key, requirement.get(key));
-                }
+                Optional.ofNullable(securitySchemes).ifPresent(_securitySchemes -> Optional.ofNullable(securitySchemes.get(key))
```

Comment:
Is this all to avoid wrapping the deleted code with an `if (securitySchemes != null)` block?

Reply:
Thanks for your comment @jimschubert I have refactored the code to better express my intentions. It's just a personal preference to use Optional.ofNullable instead of checking for null.